ocaml/utils/target_system.ml

```diff
@@ -0,0 +1,96 @@
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+type architecture =
+  | IA32
+  | X86_64
+  | ARM
+  | AArch64
+  | POWER
+  | Z
+
+let architecture () : architecture =
+  match Config.architecture with
+  | "i386" -> IA32
+  | "amd64" -> X86_64
+  | "arm" -> ARM
+  | "arm64" -> AArch64
+  | "power" -> POWER
+  | "s390x" -> Z
+  | arch -> Misc.fatal_errorf "Unknown architecture `%s'" arch
+
+type derived_system =
+  | Linux
+  | MinGW_32
+  | MinGW_64
+  | Win32
+  | Win64
+  | Cygwin
+  | MacOS_like
+  | FreeBSD
+  | NetBSD
+  | OpenBSD
+  | Generic_BSD
+  | Solaris
+  | Dragonfly
+  | GNU
+  | BeOS
+  | Unknown
+
+let derived_system () : derived_system =
+  match architecture (), Config.model, Config.system with
+  | IA32, _, "linux_aout" -> Linux
```

Comment:
```suggestion
```
I think this one is no longer supported.

Reply:
Removed